ddtrace/tracer/span_test.go

```diff
@@ -1076,3 +1076,30 @@ type stringer struct{}
 func (s *stringer) String() string {
 	return "string"
 }
+
+// TestConcurrentSpanSetTag tests that setting tags concurrently on a span directly or
+// not (through tracer.Inject when trace sampling rules are in place) does not cause
+// concurrent map writes. It seems to only be consistently reproduced with the -count=100
+// flag when running go test, but it's a good test to have.
+func TestConcurrentSpanSetTag(t *testing.T) {
```

Comment:
Yeah, if I revert the fix and run this with `-count=1` under the race detector, it passes. But if I do `-count=2`, it fails... It fails consistently if I swap the `go` statements (so the `Inject` one is first, `SetTag` is second. Strange! It also fails consistently if I do this:

```
func TestConcurrentSpanSetTag(t *testing.T) {
    testConcurrentSpanSetTag(t)
    testConcurrentSpanSetTag(t)
}

func testConcurrentSpanSetTag(t *testing.T) {
    // the actual implementation
}
```

I don't have any good ideas for why it doesn't trigger the failure the way you wrote it. But one of the things I did should hopefully catch the bug consistently without requiring `-count > 1`

Reply:
That's really curious, good catch! Although it doesn't fail consistently in my machine, it fails more often. I updated the code to reflect your suggestions. It won't be a flaky test because running it with my fix under the race detecter and `-count=100` doesn't trigger any error.